### PR TITLE
Update Parser to accept new parameters

### LINKED ISSUE #45: Update Parser to accept new parameters

### PR DESCRIPTION
**Pull request by nmtuan2001**
Currently incomplete without having the new command classes.
- [x] Finalise once all commands are ready
- [x] Add test cases to make sure Parser is working as intended
(Only added test cases for AddCommand since the other commands should be the same as the original version.)

### CODE REVIEW

**Comment by nmtuan2001** [comment]
A few comments are currently incorrect (for example, in `ApplicationDeleteCommand`, the comments mention `DeleteCommand`), but since we will eventually rename the files, I didn't change the comments.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -66,16 +66,16 @@ title: User Guide
 
 Adds an internship application to the list.
 
-Format: `add c/COMPANY_NAME d/DATE p/POSITION`
+Format: `add c/COMPANY_NAME ct/CONTACT_NUMBER d/DATE e/EMAIL p/POSITION`
 
 <div markdown="span" class="alert alert-primary">:bulb: <b>Tip:</b>
-Date must be specified in the format <em>dd/MM/yyyy</em>.
+Date must be specified in the format <em>dd-MM-yyyy</em>.
```
Comment (eesung00) [inline docs/UserGuide.md]: Date format have been changed to yyyy-MM-dd in PR #57

**Review by eesung00 — COMMENTED** [review]
LGTM 👍 just remember to fix the date format

File: src/test/java/seedu/address/testutil/TypicalApplications.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    // Manually added - Application's details found in {@code ApplicationCommandTestUtil}
+    public static final Application GOOGLE = new ApplicationBuilder().withCompany(VALID_COMPANY_GOOGLE)
+            .withContact(VALID_CONTACT_GOOGLE).withDate(VALID_DATE_GOOGLE)
+            .withEmail(VALID_EMAIL_GOOGLE).withPosition(VALID_POSITION_GOOGLE).build();
+    public static final Application FACEBOOK = new ApplicationBuilder().withCompany(VALID_COMPANY_FACEBOOK)
+            .withContact(VALID_CONTACT_FACEBOOK).withDate(VALID_DATE_FACEBOOK)
+            .withEmail(VALID_EMAIL_FACEBOOK).withPosition(VALID_POSITION_FACEBOOK).build();
+
+    public static final String KEYWORD_MATCHING_GOOGLE = "Google"; // A keyword that matches MEIER
```
Comment (teekaytai) [inline src/test/java/seedu/address/testutil/TypicalApplications.java]: This comment should be updated too
```suggestion
    public static final String KEYWORD_MATCHING_GOOGLE = "Google"; // A keyword that matches GOOGLE
```

**Review by teekaytai — COMMENTED** [review]
LGTM too! Just one minor error spotted

**Comment by jolynloh** [comment]
I think it would be great if the tests in ApplicationAddCommandParserTest can be corrected to take into account the new date format! Otherwise the UI part which relies on this PR will probably fail CI checks with failed tests

**Comment by nmtuan2001** [comment]
> I think it would be great if the tests in ApplicationAddCommandParserTest can be corrected to take into account the new date format! Otherwise the UI part which relies on this PR will probably fail CI checks with failed tests

@jolynloh Oh I updated the test with my last commit. It seems to be working fine for me. Is there any issue on your side? If you don't mind, can you help me take a screenshot?

**Comment by jolynloh** [comment]
@nmtuan2001 ah okay let me check again!

**Pull request merged**